sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -0,0 +1,198 @@
+      final ITransaction transaction =
+          hub.startTransaction(getActivityName(activity), "navigation");
+
+      // lets bind to the scope so other integrations can pick it up
+      hub.configureScope(
+          scope -> {
+            scope.withTransaction(
+                scopeTransaction -> {
+                  // we we'd not like to overwrite existent transactions bound to the Scope
+                  // manually.
+                  if (scopeTransaction == null) {
+                    scope.setTransaction(transaction);
+                  }
+                });
+          });
+
+      activitiesWithOngoingTransactions.put(activity, transaction);
+    }
+  }
+
+  private boolean isRunningTransaction(final @NonNull Activity activity) {
+    return activitiesWithOngoingTransactions.containsKey(activity);
+  }
+
+  private void stopTracing(final @NonNull Activity activity, final boolean shouldFinishTracing) {
+    if (performanceEnabled && shouldFinishTracing) {
+      final ITransaction transaction = activitiesWithOngoingTransactions.get(activity);
+      if (transaction != null) {
+        SpanStatus status = transaction.getStatus();
+        // status might be set by other integrations, let's not overwrite it
+        if (status == null) {
+          status = SpanStatus.OK;
+        }
+        //
+        transaction.finish(status);
+      }
+    }
+  }
+
+  @Override
+  public synchronized void onActivityPreCreated(
+      final @NonNull Activity activity, final @Nullable Bundle savedInstanceState) {
+    // if activity has global fields being init. and
+    // they are slow, this won't count the whole fields/ctor initialization time, but only
+    // when onCreate is actually called.
+    startTracing(activity);
+  }
+
+  @Override
+  public synchronized void onActivityCreated(
+      final @NonNull Activity activity, final @Nullable Bundle savedInstanceState) {
+    addBreadcrumb(activity, "created");
+  }
+
+  @Override
+  public synchronized void onActivityStarted(final @NonNull Activity activity) {
+    addBreadcrumb(activity, "started");
+  }
+
+  @Override
+  public synchronized void onActivityResumed(final @NonNull Activity activity) {
+    addBreadcrumb(activity, "resumed");
+  }
+
+  @Override
+  public synchronized void onActivityPostResumed(final @NonNull Activity activity) {
+    // this should be called only when onResume has been executed already, which means
+    // the UI is responsive at this moment.
+    stopTracing(activity, options.isEnableAutoActivityLifecycleTracingFinish());
+  }
+
+  @Override
+  public synchronized void onActivityPaused(final @NonNull Activity activity) {
+    addBreadcrumb(activity, "paused");
+  }
+
+  @Override
+  public synchronized void onActivityPostPaused(final @NonNull Activity activity) {
+    // in case people opt-out enableAutoActivityLifecycleTracingFinish and forgot to finish it,
+    // we do it automatically before moving to a new Activity.
+    stopTracing(activity, true);
```

Comment:
`h`: If someone gets a call after 2 seconds being on the activity, this is going to produce wrong statistics, I guess. We don't know here if someone navigates to a different activity or puts the app in the background. Maybe we need some kind of strategy as we have it already for sessions. Store it somewhere, but only end it if the user doesn't come back.

Reply:
we are doing performance, not analytics, so it's fair to finish it if not yet.
I was not finishing before and @bruno-garcia brought some uses cases that would make sense to do it (like non finished transactions, etc)

Reply:
I'm fine either way, both use cases has pros/cons, its a best effort

Reply:
consider iterating over the activity weakref list and finish pending transactions instead of finishing during onPause before creating the new Activity, this would give a bit more time to the user to finish the transaction than just finishing on onPause, this would solve the problem if the user gets a call.

Reply:
this method call should stay on onStop or onDestroy

Reply:
onDestroy because if you get a call from somebody, App goes to onStop as well